testing/test_assertrewrite.py

```diff
@@ -279,6 +279,19 @@ def f():
             assert False or 4 % 2
         assert getmsg(f) == "assert (False or (4 % 2))"
 
+    @pytest.mark.skipif("sys.version_info < (3,5)")
+    def test_at_operator_issue1290(self, testdir):
+        testdir.makepyfile("""
+            class Matrix:
+                def __init__(self, num):
+                    self.num = num
+                def __matmul__(self, other):
+                    return self.num * other.num
+
+            def test_multmat_operator():
+                assert Matrix(2) @ Matrix(3) == 42""")
+        testdir.runpytest().assert_outcomes(failed=1)
```

Comment:
Hmm doesn't this test also pass if you take out your fix in `rewrite.py`? Wouldn't be better to make the the inner test pass?